xwiki-commons-core/xwiki-commons-job/xwiki-commons-job-default/src/main/java/org/xwiki/job/internal/DefaultJobExecutor.java

```diff
@@ -262,11 +274,23 @@ public void dispose() throws ComponentLifecycleException
     // JobManager
 
     @Override
-    public Job getCurrentJob(JobGroupPath path)
+    public Job getCurrentJob(JobGroupPath groupPath)
+    {
+        // Provide a default implementation so implementors of this interface can stop implementing this method.
+        return getCurrentJobs(groupPath).stream().findFirst().orElse(null);
+    }
+
+    @Override
+    public Collection<Job> getCurrentJobs(JobGroupPath path)
     {
         JobGroupExecutor executor = this.groupExecutors.get(path);
 
-        return executor != null ? executor.currentJob : null;
+        if (executor != null) {
+            // Return an unmodifiable copy of the set of currently running jobs.
+            return Collections.unmodifiableCollection(new ArrayList<>(executor.currentJobs));
```

Comment:
It's not clear to me why you create an ArrayList. Isn't Collections.unmodifiableCollection enough to make the set readonly ?

Reply:
I want to avoid that the collection updates when jobs are started or stopped. I would expect that we expose this method in a script service and I don't want to let the author of a Velocity script deal with the fact that directly after `isEmpty()` returns `false`, there might not be any elements anymore in the collection.

We could also let this return a `List`, this might make things more explicit what to expect in terms of the different operations?

Reply:
> I want to avoid that the collection updates when jobs are started or stopped.

Very good point indeed.

> We could also let this return a List, this might make things more explicit what to expect in terms of the different operations?

Why not.

Reply:
Done. I also found and fixed some issues in the implementations for constructing the list both in the default and in the actual implementation.